File: FamilyTree_v2/FamilyTree_v2/Person.vb
Public Class Person
    'CONSTRUCTORS
    'New

    'This Is the constructor without parameters

    'Parameters:
    '    void
    'Returns:
    '    It's a contructor
    Public Sub New()
        Name = "UNKNOWN"
        FatherName = "UNKNOWN"
        MotherName = "UNKNOWN"
        FatherIndex = 0
        MotherIndex = 1
        IsAncestor = False
        Gender = "M"
    End Sub

    'CONSTRUCTORS
    'New

    'This Is the constructor with parameters

    'Parameters:
    '    String name2           This is the name of the person
    '    Integer FatherIndex2   Index of the father in the database
    '    Integer MotherIndex2   Index of the mother in the database
    '    String Gender2         Gender of the person (M/F)
    'Returns:
    '    It's a contructor
    Public Sub New(ByVal Name2 As String, ByVal FatherIndex2 As Integer, ByVal MotherIndex2 As Integer, ByVal Gender2 As String)
        Name = Name2
        FatherName = "UNKNOWN"
        MotherName = "UNKNOWN"
        FatherIndex = FatherIndex2
        MotherIndex = MotherIndex2
        IsAncestor = False
        Gender = Gender2
    End Sub

    'MEMBERS
    Public Name As String
    Public FatherName As String
    Public MotherName As String
    Public FatherIndex As Integer
    Public MotherIndex As Integer
    Public Gender As Char
    Public IsAncestor As Boolean
    Public Children As New ArrayList

    'MEMBER FUNCTIONS

    'getFullName

    'Retrieves the full name of the person, if he/she is an ancestor, then
    'we only display the last name only. Supports Mexican format of the name ONLY

    'Parameters:
    '   void
    'Returns:
    '   String FullName     Full name of the person
    Public Function getFullName() As String
        Dim FullName As String
        FullName = Name & " " & FatherName
        If IsAncestor = False Then
            FullName = FullName & " " & MotherName
        End If
        Return FullName
    End Function

    'AddChild

    'Adds a child by index from the database

    'Parameters:
    '   Integer i
    'Returns:
    '   void
    Public Sub AddChild(ByVal i As Integer)
        Children.Add(i)
    End Sub
End Class

File: MyFamilyTree_v1_1/MyFamilyTree_v1_1/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyFamilyTree_v1_1
{
	public class Person
	{
		public string name { get; set; }
		public string fatherName { get; set; }
		public string motherName { get; set; }
		public int fatherIndex { get; set; }
		public int motherIndex { get; set; }
		public char gender { get; set; }
		public List<int> children = new List<int>();

		public DateTime birthdate { get; set; }
		public int birthPlace { get; set; }
		public int birthCountry { get; set; }
		public DateTime deathdate { get; set; }
		public int deathPlace { get; set; }
		public int deathCountry { get; set; }

		public int countryResidence { get; set; }
		public int placeResidence { get; set; }

		public int university { get; set; }
		public int career { get; set; }
		public int classOf { get; set; }

		public int organization { get; set; }
		public int position { get; set; }

		public Person()
		{
			name = "Unknown";
			fatherName = "Unknown";
			motherName = "Unknown";
			fatherIndex = 0;
			motherIndex = 0;
			gender = 'M';

			birthdate = DateTime.Parse("09/09/1900");
			deathdate = birthdate;
			birthCountry = -1;
			birthPlace = -1;
			deathCountry = -1;
			deathPlace = -1;

			countryResidence = -1;
			placeResidence = -1;

			university = -1;
			career = -1;
			classOf = -1;

			organization = -1;
			position = -1;
		}

		/* isAncestor
		 * 
		 * Determines if this person is an ancestor
		 * 
		 * Paramerers:
		 *	void
		 * Returns:
		 *	bool	Whether this person is an ancestor or not
		 */
		public bool isAncestor()
		{
			return (fatherIndex == 0);
		}

		public string getFullName()
		{
			string fullName = name + " " + fatherName;
			if (!isAncestor())
			{
				fullName += (" " + motherName);
			}
			return fullName;
		}

		public string getBrokenName()
		{
			string brokenName = name + "\r\n" + fatherName;
			if (!isAncestor())
			{
				brokenName += (" " + motherName);
			}
			return brokenName;
		}

		/* addChild
		 * 
		 * Adds a new pointer to a new child for this person
		 * 
		 * Parameters:
		 *	int index	Index of the child in the population array
		 * Returns:
		 *	void
		 */
		public void addChild(int index)
		{
			children.Add(index);
		}

		// More methods are to be added here !!!
	}
}
